webcrypto/subtle_crypto.go

```diff
@@ -128,71 +130,73 @@ func (sc *SubtleCrypto) Encrypt(algorithm, key, data goja.Value) *goja.Promise {
 // The `key` parameter should be a `CryptoKey` to be used for decryption.
 //
 // The `data` parameter should contain the data to be decrypted.
-func (sc *SubtleCrypto) Decrypt(algorithm, key, data goja.Value) *goja.Promise {
+func (sc *SubtleCrypto) Decrypt( //nolint:dupl // we have two similar methods
+	algorithm, key, data goja.Value,
+) *goja.Promise {
 	rt := sc.vu.Runtime()
-	promise, resolve, reject := promises.New(sc.vu)
 
-	// 2.
-	// We obtain a copy of the key data, because we might need to modify it.
-	ciphertext, err := exportArrayBuffer(rt, data)
-	if err != nil {
-		reject(err)
-		return promise
-	}
+	var (
+		ciphertext []byte
+		ck         CryptoKey
+		decrypter  EncryptDecrypter
+	)
 
-	// 3.
-	normalized, err := normalizeAlgorithm(rt, algorithm, OperationIdentifierDecrypt)
-	if err != nil {
-		reject(err)
-		return promise
-	}
+	err := func() error {
```

Comment:
I'm not familiar with this pattern of wrapping erroring code in dedicated anonymous' function scope. Just informally, I'd be keen to hear more about its benefits 🙇🏻

Reply:
Sure :+1: 

The key point of this change is to centralize the logic of management promises. Previously, it was easier to miss that we either had to reject or return promises.  

See examples:
* https://github.com/grafana/xk6-webcrypto/pull/61#discussion_r1519621870
* https://github.com/grafana/xk6-webcrypto/pull/61#discussion_r1519623244

Also, it simplifies the operations because at this logical block, we only do operations + with returning almost vanilla errors, but it doesn't do any promise-related operations.